lib/cisco_node_utils/syslog_server.rb

```diff
@@ -42,12 +52,20 @@ def initialize(name,
 
     def self.syslogservers
       hash = {}
-
       syslogservers_list = config_get('syslog_server', 'server')
       return hash if syslogservers_list.nil?
 
       syslogservers_list.each do |id|
+        # The YAML regex isn't specific enough for some platforms,
+        # so we have to do further checking.
+        begin
+          IPAddr.new(id)
+        rescue
+          next
+        end
+
         level = config_get('syslog_server', 'level', id)
+        level = LEVEL_TO_NUM[level] if platform == :ios_xr
 
         vrf = config_get('syslog_server', 'vrf', id)
```

Comment:
Correct on 1) and 2)

Yes, I should be able to implement a) and b). 

I'll work on this tomorrow, thanks!